mysite/blog/views.py

```diff
@@ -22,3 +24,25 @@ def post_detail(request, year, month, day, post):
         publish__day=day,
     )
     return render(request, "blog/post/detail.html", {"post": post})
+
+
+def post_share(request, post_id):
+    post = get_object_or_404(Post, id=post_id, status="published")
+    sent = False
+    if request.method == "POST":
+        form = EmailPostForm(request.POST)
+        if form.is_valid():
+            cd = form.cleaned_data
+            post_url = request.build_absolute_uri(post.get_absolute_url())
+            subject = f"{cd['name']} recommends you read " f"{post.title}"
+            message = (
+                f"Read {post.title} at {post_url}\n\n"
+                f"{cd['name']}'s comments: {cd['comments']}"
+            )
+            send_mail(subject, message, "admin@myblog.com", [cd["to"]])
+            sent = True
+    else:
+        form = EmailPostForm()
+    return render(
+        request, "blog/post/share.html", {"post": post, "form": form, "sent": sent}
+    )
```

Comment:
Not clear on what this is doing. Please clean it.

Reply:
ok